lib/cassandra_migrations/railtie/initializer.rb

```diff
@@ -11,20 +11,24 @@
 
 module CassandraMigrations
 
+  if defined?(Spring)
```

Comment:
@hsgubert and @bsbodden, after looking at it more, I believe this is where the spring callback registration belongs.

I also made a safe version of restart, in case a client is not connected yet.

I've tested this with Spring and without it and it works as I expect now in both cases.

If y'all are good with the last commit's changes and the state of the PR now, I'd like to touch up the readme a bit (e.g. for forking servers like Puma), and then move forward with a merge if y'all approved of the readme tweaks.